### PR TITLE
Optimize flutter run logic for iOS by "ONLY_ACTIVE_ARCH=YES" if possible

### PR DESCRIPTION
**Pull request by kangwang1988**
This newly added logic optimize the `flutter run` logic so that when the target device is an arm64(armv7) device, it will generate products for arm64(armv7) only.
Henceforth, it will speed up the build process and doesn't create unnecessary products.

Besides, when using `flutter build ios --release`, this logic will not be triggered, and the result app is as expected.

### CODE REVIEW

**Comment by kangwang1988** [comment]
Besides, I often compile the Flutter.framework(arm64) & gen_snapshot(x86_64) and replace the original ones in flutter/bin/cache/artifacts/engine/ios directories.
Without this patch, when I run `flutter run -d xxx`
It will report 
```
Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Debug ===
    ld: warning: ignoring file
    /Users/kylewong/Codes/Flutter/alibaba-flutter/Projects/hotreload_demo/ios/Flutter/Flutter.
    framework/Flutter, file was built for arm64 which is not the architecture being linked
    (armv7):
    /Users/kylewong/Codes/Flutter/alibaba-flutter/Projects/hotreload_demo/ios/Flutter/Flutter.
    framework/Flutter
    Undefined symbols for architecture armv7:
      "_OBJC_METACLASS_$_FlutterAppDelegate", referenced from:
          _OBJC_METACLASS_$_AppDelegate in AppDelegate.o
      "_OBJC_CLASS_$_FlutterAppDelegate", referenced from:
          _OBJC_CLASS_$_AppDelegate in AppDelegate.o
    ld: symbol(s) not found for architecture armv7
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Run it from xcode directly works fine.

@cbracken

**Comment by kangwang1988** [comment]
@cbracken 
Please have a look at this pr.

image

**Comment by kangwang1988** [comment]
@cbracken 
Could you have a look at this PR?

**Comment by kangwang1988** [comment]
@cbracken 
Could you have a look at this PR?

**Review by tvolkert — COMMENTED** [review]
Is it worth optimizing the code path of `flutter run` on an iOS simulator as well?

File: packages/flutter_tools/lib/src/ios/mac.dart [hunk 1]
```diff
@@ -134,7 +134,7 @@ class IMobileDevice {
 
   Future<String> getInfoForDevice(String deviceID, String key) async {
     try {
-      final ProcessResult result = await processManager.run(<String>['ideviceinfo', '-u', deviceID, '-k', key, '--simple']);
+      final ProcessResult result = await processManager.run(<String>['ideviceinfo', '-u', deviceID, '-k', key]);
```
[the remark this answers is not in the document]
Comment (kangwang1988) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Yes, I've checked 'flutter doctor -v' and 'flutter devices' for a ios device is connected or not. They work as expected.

**Comment by tvolkert** [comment]
Is it worth optimizing the code path of `flutter run` on an iOS simulator as well?

**Comment by tvolkert** [comment]
Looks like tests are failing.

**Comment by kangwang1988** [comment]
> Looks like tests are failing.

Resulted from ` info • Prefer const with constant constructors • dev/benchmarks/macrobenchmarks/lib/src/cull_opacity.dart:16:62 • prefer_const_constructors`,
am I supposed to fix it?

**Comment by kangwang1988** [comment]
> Is it worth optimizing the code path of `flutter run` on an iOS simulator as well?

No, there is an existing logic doing the job:
```
  if (buildForDevice) {
    buildCommands.addAll(<String>['-sdk', 'iphoneos']);
  } else {
    buildCommands.addAll(<String>['-sdk', 'iphonesimulator', '-arch', 'x86_64']);
  }
```

**Comment by tvolkert** [comment]
> It has nothing to do with this pr, am I supposed to fix it?

I think you need to sync and rebase.

**Comment by kangwang1988** [comment]
Close this pr because of improper rebase operation and track it in https://github.com/flutter/flutter/pull/27817

**Pull request closed without merging**